src/app/@ansyn/ansyn/app-effects/effects/map.app.effects.ts

```diff
@@ -153,7 +154,19 @@ export class MapAppEffects {
 	onDisplayOverlay$: ObservableInput<any> = this.actions$
 		.ofType<DisplayOverlayAction>(OverlaysActionTypes.DISPLAY_OVERLAY)
 		.withLatestFrom(this.store$.select(mapStateSelector))
-		.filter(([{ payload }]: [DisplayOverlayAction, IMapState]) => OverlaysService.isFullOverlay(payload.overlay))
+		.filter(([{action, payload }]: [any, any]) => OverlaysService.isFullOverlay(payload.overlay))
+		.do(([action, payload]: [any, any]) =>
+		{
+			if (payload.mapsList.some((aMap) => {
+				if ((action.payload.overlay.id === undefined) || (aMap.data.overlay === undefined)) {
```

Comment:
if `aMap.data.overlay === null` it's ok ?
if action.payload.overlay not null its ok to assume that id exist...
```typescript
Boolean(action.payload.overlay && aMap.data.overlay)
```

finally:
```typescript
const existingOverlays = Boolean(action.payload.overlay && aMap.data.overlay);
const equalOverlays = aMap.data.overlay.id === action.payload.overlay.id;
return existingOverlays && equalOverlays ;
```

Reply:
won't work, aMap.data.overlay.id === action.payload.overlay.id; returns causing crash when aMap.data.overlay.id or action.payload.overlay.id are undefined.